### ISSUE TITLE
Scan() command ignores start_key and end_key

### ISSUE THREAD

**Issue by PaulGrandperrin** [tikv/client-rust#127]
I ran the raw API example code on a fresh cluster and I get an assertion error here:
https://github.com/tikv/client-rust/blob/master/examples/raw.rs#L95-L97
```
Put key "TiKV", value "Rust".
Get key `TiKV` returned value Some(Value("Rust")).
Key: `TiKV` deleted
Found values: [KvPair(6B31, "v1"), KvPair(6B32, "v2")] for keys: [Key(6B31), Key(6B32)]
thread 'main' panicked at 'assertion failed: `(left == right)`
  left: `[Key(6B31), Key(6B32), Key(6B33)]`,
 right: `[Key(6B31), Key(6B32)]`', src/main.rs:101:5
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace.
```

So I added two `println!` in `src/raw/requests.rs` to get more info about when those start and end keys where lost, here is the output:
```
Put key "TiKV", value "Rust".
Get key `TiKV` returned value Some(Value("Rust")).
Key: `TiKV` deleted
Found values: [KvPair(6B31, "v1"), KvPair(6B32, "v2")] for keys: [Key(6B31), Key(6B32)]
>>> in new_raw_scan_request : req=RawScanRequest { context: None, start_key: [107, 49], limit: 10, key_only: true, cf: "", reverse: false, end_key: [107, 50, 0] }
>>> in make_rpc_request: req=RawScanRequest { context: Some(Context { region_id: 2, region_epoch: Some(RegionEpoch { conf_ver: 5, version: 1 }), peer: Some(Peer { id: 7, store_id: 5, is_learner: false }), term: 0, priority: Normal, isolation_level: Si, not_fill_cache: false, sync_log: false, handle_time: false, scan_detail: false, replica_read: false, resolved_locks: [], max_execution_duration_ms: 0, applied_index: 0 }), start_key: [], limit: 10, key_only: true, cf: "", reverse: false, end_key: [] }
thread 'main' panicked at 'assertion failed: `(left == right)`
  left: `[Key(6B31), Key(6B32), Key(6B33)]`,
 right: `[Key(6B31), Key(6B32)]`', src/main.rs:101:5
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace.
```
We can see that in the first object the `start_key`, `end_key`, and `limit` fields are correctly set, but in the second object, `start_key` and `end_key` became empty but `limit` kept its correct value (10).

**Comment by Snnappie** [comment]
I believe I have a fix for this. I can open a pull request soon.

**Comment by ekexium** [comment]
Fixed by #164. Note the assertion is mistakenly changed (probably to pass the CI) in #134. #178 fix the example.
